modules/aws/eks/eks.tf

```diff
@@ -51,6 +54,63 @@ resource "aws_eks_cluster" "this" {
     }
   }
 
+  tags       = local.global_tags
+  depends_on = [aws_cloudwatch_log_group.this]
+}
+
+
+
+# CloudWatch Encrytion
+resource "aws_kms_key" "cloudwatch_encryption" {
+  description         = "Used for cloudwatch encryption"
+  enable_key_rotation = true
+
+  policy = <<EOF
+{
+  "Version" : "2012-10-17",
+  "Id" : "key-default-1",
+  "Statement" : [ {
+      "Sid" : "Enable IAM User Permissions",
+      "Effect" : "Allow",
+      "Principal" : {
+        "AWS" : "arn:aws:iam::${data.aws_caller_identity.current.account_id}:root"
```

Comment:
I think `data.aws_caller_identity.current.account_id` is the Terraform user's account, whereas according to  https://docs.aws.amazon.com/kms/latest/developerguide/key-policy-default.html it should be the account in which we are creating the key?

Reply:
As it turns out, `data.aws_caller_identity.current.account_id` will be correct because there is no way to act in account 2 with credentials from account 1 except by assuming a role in account 2.

In this example, `sandbox` is a sub-account in my private organization. In the second step, aws-vault implicitly assumes a role and executes a command:
```sh
$aws-vault exec bittrance -- aws sts get-caller-identity
{
    "UserId": "AIDA<ID>",
    "Account": "<account-1>",
    "Arn": "arn:aws:iam::<account-1>:user/bittrance"
}

$ aws-vault exec sandbox -- aws sts get-caller-identity
{
    "UserId": "AROA<ID>:1646383552799375356",
    "Account": "<account-2>",
    "Arn": "arn:aws:sts::<account-2>:assumed-role/OrganizationAccountAccessRole/1646383552799375356"
}
```